## Getting familar with dataframe and pandas Day 1.

### In below cell I have import csv a Comma Seperated Value and Pandas library. Panda library is used for data manipulation.
read_csv helps to read df in form of csv format. 
If we want to read json or any file format just repalce csv with corresponding file format. Like to read jso file use read_json("path of file to be read")

In [ ]:


import csv
import pandas as pd
df = pd.read_csv('quarteroverseas.csv')
df.head()

### Use of head and tail
head=df.head() give top 5 values
tail=pf.tail() give top last 5 values


In [3]:
df.shape

(100523, 13)

### Use of shape
shape function is used to get the number of rows and cols that are present in a dataframe
the value in 0 index = no of rows
the value in 1 inder = no of cols
In above we can see that there are 100523 rows and 13 columns

In [4]:
df.columns

Index(['Series_reference', 'Period', 'Data_value', 'STATUS', 'UNITS',
       'MAGNTUDE', 'Subject', 'Group', 'Series_title_1', 'Series_title_2',
       'Series_title_3', 'Series_title_4', 'Series_title_5'],
      dtype='object')

In [5]:
df['STATUS']

0               FINAL
1               FINAL
2               FINAL
3               FINAL
4               FINAL
             ...     
100518        REVISED
100519        REVISED
100520        REVISED
100521        REVISED
100522    PROVISIONAL
Name: STATUS, Length: 100523, dtype: object

### Use of columns and how to read  data from column/s
columns fxn is used to get labels that are associated with columns name
if we want the value of column then df['colunm_name'] can be use. But when we try to use same for multiple cols then it will give rise to index error.So we will learn about loc and iloc in Day 2.

# Creating own dataframe
I will be using 2 method for creating dataframe one will be list of dictionary and another will be list of list say 2d list 

### Dataframe using list of dictionary
At first create clo-name to be used and make list of dictionary of data

In [11]:
col_name=['Name', 'Rollno', 'Address']
data_dict = [{'Name':'Ram', 'Rollno' : 25, 'Address' : 'Kathmandu'},{'Name':'Sam', 'Rollno' : 5, 'Address' : 'Patan'},{'Name':'Sita', 'Rollno' : 5, 'Address' : 'Bhaktapur'}]

#### Use of DictWriter fxn
open("name of file to create dataframe",'w'=write operation ) as =assignment operation to assign it as variable name here it is csvfile
make use of DictWriter to make dataframw
at first always write column name so make use of writeheader() fxn
and then write rows in a loop

In [18]:
with open ('dataset.csv','w') as csvfile:
    writer = csv.DictWriter(csvfile, fieldnames = col_name)
    writer.writeheader()
    for d in data_dict:
        writer.writerow(d)
 
        

### just reading the dataframe created in cell 18

In [19]:
df= pd.read_csv('dataset.csv')
df

Name  Rollno    Address
0   Ram      25  Kathmandu
1   Sam       5      Patan
2  Sita       5  Bhaktapur

### Here i have made dataframe by using list of list 

In [22]:
dataset=[['Name','Age','Rollno'],['Ram', '25','1'], ['Sita','22','3'],['Gita','24','4']]

file=open('dataset_list.csv','w')
x=csv.writer(file)
x.writerows(dataset)
file.close()

In [23]:
df1=pd.read_csv('dataset_list.csv')
df1

Name  Age  Rollno
0   Ram   25       1
1  Sita   22       3
2  Gita   24       4